### DISCUSSION TITLE
Is it possible to pass optional parameters in the routing for the edit page?

### DISCUSSION THREAD

**Discussion by bzy107** [filamentphp/filament#9351] [Help] [answered]
I want to provide parameters like this:

```php
public static function getPages(): array
{
    ...
    'edit' => Pages\EditItem::route('/{record}/edit/{parameter}'),
}
```

`{parameter}` are expected to be values of the record's columns.

**Comment by sinnbeck** [comment]
Can you explain what you need it for? The record already has all data from its columns?

To do it you would need to also change the EditAction url.
```php
EditAction::make()
                    ->url(fn ($record): string => self::getUrl('edit', ['record' => $record, 'parameterfoo' => 'foobar']))
```
And add `public $parameter` to the EditResource

**Reply by sinnbeck** [reply, marked as answer]
Be aware that if it should be optional, you need to postfix it with ? https://laravel.com/docs/10.x/routing#parameters-optional-parameters
```php
public static function getPages(): array
{
    ...
    'edit' => Pages\EditItem::route('/{record}/edit/{parameter?}'),
}

**Reply by bzy107** [reply]
Thank you.
I was able to add the parameters.

The intention of my question is that I want to edit a table with a composite primary key, but I don't know how to pass two keys, so I decided to pass them as parameters. Please let me know if there is a better way.

**Reply by sinnbeck** [reply]
Ah ok. No, not if there isnt some identifier in session. For instance, tenent ID is handled as a global scope and is taken from session

If the question is answered, remember to mark and answer

**Reply by bzy107** [reply]
thanks.

**Comment by jimiero** [comment]
Hello,

I have my custom filament page, and I want it to be able to receive 2 parameters in the slug, the second I want to be optional so I've set this:

 protected static ?string $slug = 'centralize/{year}/{month?}';
 
 how should my mounth method look like because if I do:

```php
      public function mount(int $year, $month): void
    {
```
i get this error:

Unable to resolve dependency [Parameter #1 [ <required> $luna ]] in class App\Filament\Pages\Centralizer
